parquet-column/src/main/java/org/apache/parquet/column/statistics/BinaryStatistics.java

```diff
@@ -18,13 +18,29 @@
  */
 package org.apache.parquet.column.statistics;
 
+import org.apache.parquet.column.Comparators;
 import org.apache.parquet.io.api.Binary;
+import org.apache.parquet.schema.OriginalType;
+import org.apache.parquet.schema.PrimitiveType;
 
 public class BinaryStatistics extends Statistics<Binary> {
 
+  private final Comparators.BinaryComparator comparator;
   private Binary max;
   private Binary min;
 
+  /**
+   * @deprecated Use {@link Statistics#getStatsBasedOnType(PrimitiveType.PrimitiveTypeName, OriginalType)} instead
+   */
+  @Deprecated
+  public BinaryStatistics() {
+    this(PrimitiveType.PrimitiveTypeName.BINARY, null);
+  }
+
+  BinaryStatistics(PrimitiveType.PrimitiveTypeName type, OriginalType logicalType) {
```

Comment:
I think I mentioned this earlier, but I think it would be a bit cleaner to pass the `Type` instance and use that to get a comparator and coerce to strings.